src/python/pants/jvm/resolve/coursier_setup.py

```diff
@@ -80,7 +80,7 @@
         --json-output-file="$json_output_file" \
         "${{@//{coursier_working_directory}/$working_dir}}"
     /bin/mkdir -p classpath
-    {python_path} {coursier_bin_dir}/coursier_post_processing_script.py "$json_output_file"
+    '{python_path}' {coursier_bin_dir}/coursier_post_processing_script.py "$json_output_file"
```

Comment:
@cognifloyd this did the trick for you then?

Reply:
Yes. I tested this before I made the post in slack. I was concerned that there could be more variation in the path including a `'` char which would require additional escaping.

Since this is using `uname -mps`, as you pointed out, I googled for examples of the output others get from that command. I'm pretty sure the verbose processor description is due to my using Gentoo with a kernel customized specifically for my processor instead of being a generic x86_64 kernel. So, I'm already in a very unusual / edge case setup, and a `'` is a vanishingly unlikely possibility, so I'm satisfied that this is the best fix and that additional escaping complexity is not necessary.

Reply:
Ok. I think the fully robust way to quote here is to instead `shlex.quote` the input data to the formatted string. That should handle all manner of oddness.

Reply:
@cognifloyd I'm holding back on a merge for your fix or pushback on using `shlex.quote`.

Reply:
I switched to `shlex.quote`. my Gentoo box is not available atm so I haven't tested yet.

Reply:
Ok, great. Thanks @cognifloyd. I'll wait for that manual test since there is no unit test coverage for this.

Reply:
OK. I tested this, and it works for me.